src/components/form-draft/publish.vue

```diff
@@ -34,7 +34,39 @@ except according to the terms contained in the LICENSE file.
       <div class="modal-introduction">
         <p>{{ $t('introduction[0]') }}</p>
         <p>{{ $t('introduction[1]') }}</p>
+
+        <template v-if="formDraft.entityRelated && hasDatasetDiff">
+          <hr>
+          <i18n-t tag="p" keypath="dataset.introduction.full">
+            <template #inAddition>
+              <strong>{{ $t('dataset.introduction.inAddition') }}</strong>
+            </template>
+          </i18n-t>
+          <ul class="dataset-list">
+            <template v-for="dataset of formDraftDatasetDiff" :key="dataset.name">
+              <i18n-t v-if="dataset.isNew" tag="li" keypath="dataset.newDataset">
+                <template #datasetName>
+                  <strong>{{ dataset.name }}</strong>
+                </template>
+              </i18n-t>
+              <template v-for="property of dataset.properties" :key="property.name">
+                <i18n-t v-if="property.isNew" tag="li" keypath="dataset.newProperty">
+                  <template #datasetName>
+                    <strong>{{ dataset.name }}</strong>
+                  </template>
+                  <template #propertyName>
+                    <strong>{{ property.name }}</strong>
+                  </template>
+                </i18n-t>
+              </template>
+            </template>
+          </ul>
+        </template>
+
+        <hr v-if="draftVersionStringIsDuplicate">
         <p v-if="draftVersionStringIsDuplicate">{{ $t('introduction[2]') }}</p>
+
+        <p>{{ $t('introduction[3]') }}</p>
```

Comment:
This is the line that says "Would you like to proceed?" but I think it's meant to go within the dataset diff block above.

Currently, it shows up at the bottom of all publish modals, which isn't terrible, it just makes the publish modal longer than it needs to be, and it can get pretty long. Also, it also winds up appearing between the message about needing a new form version, and the form version input. Moving it up to the conditional dataset block will help!

<img width="625" alt="Screen Shot 2022-11-14 at 3 34 47 PM" src="https://user-images.githubusercontent.com/76205/201790092-2a1f4905-c997-4d79-838b-41342436b03c.png">